usaspending_api/common/api_request_utils.py

```diff
@@ -464,12 +466,16 @@ def build_response(self):
                     "matched_ids": Location.objects.filter(Q(**{"location_congressional_code": row[0], "location_state_code": row[1], "location_state_name": row[2]})).values_list("location_id", flat=True)
                 }
                 response_object.append(response_row)
+                if len(response_object) >= limit:
```

Comment:
After toying with it for a while, I don't think it will work. This is a case where we're not returning an actual model from the DB, but a custom response so we kind of have to do it this way. Unless we want to make some new "hierarchical geography" model and a new endpoint for that.

I also think it is probably faster to do `SELECT ids WHERE blah` vs. trying to aggregate the id's together when executing a distinct, which is what we'd have to do; unless I've missed some nifty django function.